src/xtgeo/well/_well_aux.py

```diff
@@ -0,0 +1,95 @@
+"""Auxillary functions for the well class
+
+'self' is a Well() instance
+
+"""
+
+from __future__ import annotations
+
+import functools
+import warnings
+from pathlib import Path
+from typing import Callable, Optional
+
+import pandas as pd
+
+import xtgeo
+from xtgeo.common import XTGeoDialog
+
+from ..xyz_common._xyz_enum import _AttrName
+from . import _well_io
+
+xtg = XTGeoDialog()
```

Comment:
move to allow_deprecated_init(...) ? Would still run import time, but we connect it to the function. If not, might want to add annotation: Final?

Reply:
Not sure what you mean here, can you eloborate?